pkg/agent/flannel/setup.go

```diff
@@ -133,7 +134,11 @@ func createFlannelConf(nodeConfig *config.Node) error {
 			return err
 		}
 	case config.FlannelBackendWireguard:
-		backendConf = wireguardBackend
+		wgAddPath, err := exec.LookPath("wg-add.sh")
+		if err != nil {
+			return err
+		}
+		backendConf = strings.Replace(wireguardBackend, "wg-add.sh", wgAddPath, -1)
```

Comment:
It feels like doing a strings.Replace is a bad idea, compared to passing the PATH in the env. I see that change mentioned, but I am curious, of why the change in approach?

Reply:
This looks honestly like a better idea to me, meaning, it seems better to use an absolute path when known that altering the PATH environment in the flannel project extension backend. From their perspective it could make very little sense to set PATH if that would already be inherited from the parent process environment anyway.
And `strings.Replace` is already being used in a similar way in the ipsec backend setup and also to set the overall backend configuration in the flannel config json, is there any particular issue with it that I am unaware of?

Reply:
@erikwilson Let me know if this makes sense. Otherwise we can attempt to go for the original approach.

Reply:
I think it is an unintentional bug in flannel for the extension backend to replace rather than append to the current env. 
Considering the ipsec backend, the charon handler does not replace the env and therefore inherits from the process.

If an upstream flannel change is not possible we should at a minimum be replacing templated variables, and I would prefer a templated path with something better suited to k3s pre-packaged user space, for example `export PATH=%PATH%; ...` in each of the commands.

Reply:
Ok! let's try this upstream first.

Reply:
This is it.
https://github.com/coreos/flannel/pull/1404